xbmc/platform/darwin/PlatformDarwin.cpp

```diff
@@ -22,3 +22,12 @@ bool CPlatformDarwin::InitStageOne()
 
   return true;
 }
+
+bool CPlatformDarwin::SupportsUserInstalledBinaryAddons()
+{
+#if defined(TARGET_DARWIN_EMBEDDED)
+  return false;
+#else
+  return CPlatform::SupportsUserInstalledBinaryAddons();
+#endif
+}
```

Comment:
thexai actually raises a good point. Dont put this in here (this is global for all darwin platforms), instead put the platform specific stuff in their own stuff

osx - https://github.com/xbmc/xbmc/blob/master/xbmc/platform/darwin/osx/PlatformDarwinOSX.cpp
darwinembedded - https://github.com/xbmc/xbmc/blob/master/xbmc/platform/darwin/ios-common/PlatformDarwinEmbedded.cpp

No need for an ifdef inside those locations. osx can just return true (or the superclass call), return false in the darwinembedded, and then if @sy6sy2 believes theres a need to differentiate (in another PR) between jailbreak/no jb, its handled without mixing shit into common code.

Reply:
IMO we don't need a difference between JB and no JB users concerning this stuff.
IIRC we do not provide iOS/tvOS binary addons on our repo ATM, right? So even JB users can't install binary addons from repo because we don't have binary addons on repo.
Maybe it's something that will be useful later if we (for example) deliver some games add-ons on repo for iOS/tvOS, then may be JB user will be able to download them from repo and run it, but I need to test that before to be sure that binary add-ons are able to run outside the `framework` folder of the mother application.
So we can put JB and no JB user in the same case ATM.

Reply:
@fuzzard thanks. Changed.